docs/DeveloperGuide.md

```diff
@@ -441,43 +441,107 @@ considering code quality and end user experience.
-The `UpdateCommandParser` implements `Parser` to parse the user input, the `UpdateCommand` extends `Command` to execute the main logic,
-and the `UpdateProductDescriptor` allows the parser to pass a specification of the updated product to the `UpdateCommand`.
+#### What the update product feature does
+The update product feature allows users to modify the quantity or tags of existing products listed under suppliers and warehouses, or add new products to their existing lists.
+
+
+#### How it is implemented
+The update product mechanism is facilitated by 3 major components: `UpdateCommandParser`, `UpdateCommand`,  and the `UpdateProductDescriptor`.
+`UpdateCommandParser`'s job is to parse the user input to generate the correct objects for the `UpdateCommand`.
+`UpdateCommand`'s job is to execute the main logic for generating the updated list of products for the model.
+`UpdateProductDescriptor`'s job is to serve as a medium to allow the `UpdateCommandParser` to pass a specification of the updated product to the `UpdateCommand`.
 
-Given below is an example usage scenario and how the update product mechanism behaves at each step.
+Given below is an example usage scenario, together with a sequence diagram, to show how the update product mechanism behaves at each step.
+![Update Product Command Sequence Diagram](images/UpdateCommandSequenceDiagram.png)
 
 Step 1. The user decides to update the stock for a product called 'Panadol' with a new quantity of 50 units
 in the warehouse named 'Jurong Warehouse'. The user also decides that he wants to give 'Panadol' a tag 'fever'.
 The user does this by executing the `update ct/w n/Jurong Warehouse pd/Panadol q/50 t/fever` command.
-The `ClinicParser#parseCommand` will then call the `UpdateCommandParser#parse` method with all the arguments
-passed by the user.
+The input string will then be passed to the `UpdateCommandParser`.
 
-Step 2. `UpdateCommandParser#parse` then attempts to create new instances of `name` for the supplier/warehouse
-and the product, and a new `UpdateProductDescriptor` with the provided quantity and tags, if any. An exception will be thrown if any of the arguments are invalid, which will be presented on the GUI.
-After which, it will call the `UpdateCommand` with the `Type`, warehouse/supplier's `Name` and `UpdateProductDescriptor` created,
-and return it to `ClinicParser#Parse` which will in turn return the `UpdateCommand` to `LogicManager#execute`.
+Step 2. By matching the prefixes provided, `UpdateCommandParser#parse` then attempts to create new instances of `Name` for the supplier/warehouse
+and the product, and a new `UpdateProductDescriptor` with the provided quantity and tags, if any. An exception will be thrown if any of the arguments are invalid, or if the type and names of the supplier/warehouse and product are not supplied, where an error message will be presented on the GUI.
```

Comment:
I think would be clearer if the line is changed to "`Name` for supplier/warehouse and product as well as a new `UpdateProductDescriptor` with the provided quantity and tags, if any."

For the second part of the sentence, maybe we can split it up to:
or if the type and names of the supplier/warehouse and product are not supplied. Here, an error message will be presented on the GUI.